src/plugins/intel_npu/src/plugin/npuw/lazy_tensor.hpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <memory>
+#include <mutex>
+#include <tuple>
+#include <unordered_map>
+#include <variant>
+
+#include "logging.hpp"
+#include "openvino/runtime/make_tensor.hpp"
+#include "openvino/runtime/tensor.hpp"
+#include "util.hpp"
+
+namespace ov {
+namespace npuw {
+namespace weights {
+
+enum class TransformType : int { TENSOR, PERMUTE, CONVERT, CONCAT };
```

Comment:
THIS should be the variant. What is `LTData` in your case?

Reply:
Renamed `TESNOR` -> `THIS` and `LTData` -> `OrigData`.
The idea is to have `ov::Tensor` or `Constant` as the original data (we need both since both cases are utilized in the code).
`CONCAT` is a special case - I support the logic that `CONCAT` doesn't have parent (no `OrigData`) and instead it has a vector of `LazyTensor` which will be concatenated on call to form that original data